base/sparse/cholmod.jl

```diff
@@ -65,46 +67,49 @@ const version = VersionNumber(version_array...)
 function __init__()
     try
         ### Check if the linked library is compatible with the Julia code
-        if Libdl.dlsym(Libdl.dlopen("libcholmod"), :cholmod_version) == C_NULL
-            hasversion = false
+        hasversion = false
+        tmp = Array(Cint, 3)
+
+        # CHOLMOD < 2.1.0 does not include cholmod_version()
+        try
+            ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), version_array)
+            ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), tmp)
+            hasversion = true
+        end
+
+        if !hasversion || VersionNumber(version_array...) < CHOLMOD_MIN_VERSION
+		    warn("""
+
+		        CHOLMOD version incompatibility
+
+		        Julia was compiled with CHOLMOD version $version. It is
+		        currently linked with a version older than
+		        $(CHOLMOD_MIN_VERSION). This might cause Julia to
+		        terminate when working with sparse matrix factorizations,
+		        e.g. solving systems of equations with \\.
+
+		        It is recommended that you use Julia with a recent version
+		        of CHOLMOD, or download the OS X or generic Linux binaries
```

Comment:
Could we just say "binaries from www.julialang.org" without mentioning the OS here? MSYS2 may eventually have a Julia package where this could also apply.